Controller/CRUDController.php

```diff
@@ -1298,6 +1281,43 @@ protected function getCsrfToken($intention)
     }
 
     /**
+     * Hook to handle access authorization.
+     *
+     * @param string $action
+     * @param object $object
+     */
+    protected function checkAccess($action, $object = null)
+    {
+        $access = array_merge(array(
+            'acl'                     => 'MASTER',
+            'export'                  => 'EXPORT',
+            'historyCompareRevisions' => 'EDIT',
+            'historyViewRevision'     => 'EDIT',
+            'history'                 => 'EDIT',
+            'edit'                    => 'EDIT',
+            'show'                    => 'VIEW',
+            'create'                  => 'CREATE',
+            'delete'                  => 'DELETE',
+            'batchDelete'             => 'DELETE',
+            'list'                    => 'LIST',
+        ), $this->accessMapping);
+
+        if (!array_key_exists($action, $access)) {
```

Comment:
It looks more like `InvalidArgumentException` here or `LogicException` - that something is missconfigured.